examples/modular-drivers/src/moisture_sensor.rs

```diff
@@ -25,11 +22,11 @@ pub fn register_models(registry: &mut ComponentRegistry) -> Result<(), RegistryE
 }
 
 #[derive(DoCommand)]
-pub struct MoistureSensor {
-    reader: AnalogReaderType<u16>,
+pub struct MoistureSensor<T: AnalogReader<u16, Error = AnalogError>> {
```

Comment:
IIUC, this is now needed because clippy is worked up. It concludes that it doesn't know if `AnalogReaderType` is `Send` because of the `dyn AnalogReader`.

Would it just be easier to declare `AnalogReaderType` as `pub type AnalogReaderType<W, E = AnalogError> = Arc<Mutex<dyn AnalogReader<W, Error = E> + Send>>;`?

Reply:
yes, but I was already doing that to Sensor and I wanted to keep the PR small